core/src/test/scala/cats/derived/adtdefns.scala

```diff
@@ -77,6 +77,18 @@ object TestDefns {
       Cogen[(Int, T, Long, List[T], String)].contramap(unapply(_).get)
   }
 
+  case class Bivariant[A](run: A => Boolean, store: A)
```

Comment:
Hmm, this is not really bivariant. Bivariance (also called sometimes "phantom variance") is an interesting concept: https://typelevel.org/blog/2016/09/19/variance-phantom.html. It can be both covariant and contravariant at the same time. Anyway just wanted to mention that, it's just a test class.